op-node/rollup/driver/driver.go

```diff
@@ -172,13 +172,14 @@ func NewDriver(
 	syncCfg *sync.Config,
 	sequencerConductor conductor.SequencerConductor,
 	plasma PlasmaIface,
+	engineClientKind engine.Kind,
```

Comment:
Would it not be better to hide the "engine kind" from the main logic, and instead translate the engine-kind CLI flag into config values in the `sync.Config`? A `SupportsPostFinalizationELSync` config attribute in the `sync.Config` makes more sense than reading potentially conflicting sync-configuration from two different objects.

Reply:
Thanks! I agree with you, and it makes sense since `SupportsPostFinalizationELSync` indeed is a configuration for the syncing behavior. I updated the code to use the engine kind from the flags and feed it directly into the `sync.Config`: https://github.com/ethereum-optimism/optimism/pull/10767/commits/ac208ef3fef13a3340ec0391cbb17628dae420b4